File: ContactListMvvmCross/ContactListMvvmCross/Test3.cs
using System.IO;
using System.Text;
using Android.App;
using Android.Runtime;

namespace ContactListXamarin
{
    public class Test3 : ITest
    {
        private readonly Activity _activity;

        public Test3(Activity activity)
        {
            _activity = activity;
        }

        public void Run()
        {
            string result = "";
            var builder = new StringBuilder();
            for (int i = 0; i < 100000; i++) {
                builder.Append(i);
            }
            result = builder.ToString();
            result = result.Substring(1000);
            result = result.Replace("10", "");
            bool containsSubstring = result.Contains("123");
            string[] parts = result.Split('2');
            int length = parts.Length;
        }
        
        //private byte[] LoadData(string filename)
        //{
        //    var stream = _activity.Assets.Open (filename) as InputStreamInvoker;
        //    var size = stream.BaseInputStream.Available();
        //    var bytes = new byte[size];
        //    stream.Read (bytes, 0, bytes.Length);
        //    stream.Close ();
        //    return bytes;
        //}
    }
}

File: ContactListJava/app/src/main/java/com/example/bogat/contactlistjava/Test3.java
package com.example.bogat.contactlistjava;

import android.app.Activity;

/**
 * Created by bogat on 3/5/2018.
 */

public class Test3 implements Test {
    private Activity activity;

    public Test3(Activity activity) {
        this.activity = activity;
    }

    @Override
    public void Run() {
        String result = "";
        StringBuilder builder = new StringBuilder();
        for (int i = 0; i < 100000; i++) {
            //result += String.valueOf(i); -- this performs 3x times slower than in xamarin O_o
            builder.append(i);
        }
        result = builder.toString();
        result = result.substring(1000); //don't know how it works but if it just sets offset to 1000 it will be cheating ;)
        result = result.replaceAll("10", "");
        boolean containsSubstring = result.contains("123");
        String[] parts = result.split("2");
        int length = parts.length;
    }

    /*private byte[] LoadData(String inFile) {
        byte[] buffer = new byte[0];
        try {
            InputStream stream = activity.getAssets().open(inFile);
            int size = stream.available();
            buffer = new byte[size];
            stream.read(buffer);
            stream.close();
        } catch (IOException e) {
            // Handle exceptions here
        }
        return buffer;
    }*/
}